client/src/Pages/User/Cart/index.js

```diff
@@ -1,352 +1,582 @@
+    pickupObjs.unshift({value: "ASAP", label: "ASAP"});
+  }
+  return pickupObjs;
 }
 
-const computeAvailableMinutes = (
-  hr,
-  startHour,
-  startMinute,
-  endHour,
-  endMinute
-) => {
-  const hour = moment().hour()
-  const minute = moment().minute()
-  let start = 0
-  let end = -1
-  if (hr == startHour) {
-    if (hr == hour) {
-      end = Math.max(minute, startMinute)
-    } else {
-      end = startMinute
-    }
-  } else if (hr == endHour) {
-    start = endMinute
-    end = 59
-  } else if (hr == hour) {
-    end = minute
+function calculateNextHours(
+  currHour,
+  currMinute,
+  startHours,
+  startMinutes,
+  endHours,
+  endMinutes
+) {
+  const currTime = currHour + currMinute / 60;
+  const endTimes = [];
+  for (let i = 0; i < endHours.length; i++) {
+    endTimes.push(endHours[i] + endMinutes[i] / 60);
+  }
+  const startTimes = [];
+  for (let i = 0; i < startHours.length; i++) {
+    startTimes.push(startHours[i] + startMinutes[i] / 60);
   }
-  let i = start
-  const rtn = []
-  while (i <= end) {
-    rtn.push(i)
-    i += 1
+  let idx = 0;
+  while (currTime >= startTimes[idx]) {
+    idx += 1;
   }
-  return rtn
+  let timeIntervals = [];
+  let newIdx = 0;
+  //When restaurant is closed for the day
+  if (idx === endTimes.length && currTime >= endTimes[idx - 1]) {
+    return [[0, 0, 0, 0]];
+  }
+  //When restaurant is not open currently for orders, but open later in the day
+  else if (
+    idx === 0 ||
+    (idx > 0 && currTime >= endTimes[idx - 1]) ||
+    currTime >= endTimes[idx - 1] - 0.25
+  ) {
+    timeIntervals.push([
+      startHours[idx],
+      startMinutes[idx],
+      endHours[idx],
+      endMinutes[idx],
+    ]);
+    newIdx = idx + 1;
+  }
+  //When the restaurant is currently open for orders
+  else {
+    timeIntervals.push([
+      currHour,
+      currMinute,
+      endHours[idx - 1],
+      endMinutes[idx - 1],
+    ]);
+    newIdx = idx;
+  }
+  while (newIdx < endTimes.length) {
+    timeIntervals.push([
+      startHours[newIdx],
+      startMinutes[newIdx],
+      endHours[newIdx],
+      endMinutes[newIdx],
+    ]);
+    newIdx += 1;
+  }
+  return timeIntervals;
+}
+
+function useIdle(options) {
+  const [isIdle, setIsIdle] = useState(false);
+  useEffect(() => {
+    const activityDetector = createActivityDetector(options);
+    activityDetector.on("idle", () => {
+      setIsIdle(true);
+    });
+    // activityDetector.on('active', ()=>{setIsIdle(false)});
+    return () => {
+      activityDetector.stop();
+    };
+  }, []);
+  return isIdle;
 }
 
-function CartDetail () {
-  const [totals, setTotals] = useState(defaultTotals)
-  const [pickupTime, setPickupTime] = useState(null)
-  const [nullError, setNullError] = useState(checkNullFields())
+function CartDetail() {
+  const [totals, setTotals] = useState(defaultTotals);
+  const [pickupTime, setPickupTime] = useState(null);
+  const [paymentMethod, setPaymentMethod] = useState("CREDIT");
```

Comment:
Set initial state of payment Method as null, so later we can do a null check to see whether or not a user actually selected a payment method (this null check also needs to be implemented)